pkg/csource/csource_test.go

```diff
@@ -24,7 +24,7 @@ func TestGenerate(t *testing.T) {
 	t.Parallel()
 	for _, target := range prog.AllTargets() {
 		switch target.OS {
-		case "netbsd", "windows":
+		case "netbsd", "windows", "freebsd":
```

Comment:
But this actually passes for me on linux (and on the buildbot). I would prefer to not lose so hard earned test coverage.
I think we need something like:
```
case "freebsd":
    if runtime.GOOS != "linux" {
        continue
    }
```

Reply:
Agree, that's better.

Reply:
By looking closer to this point, I don't get it. If we change in executor/common.h

#include <endian.h>

to

#if GOOS_freebsd && !__linux__
#include <sys/endian.h> // for htobe*.
#else
#include <endian.h> // for htobe*.
#endif

we cannot do test on linux when the test program is predending to run on freebsd. At least, when I change the switch-case in csource_test.go to

switch target.OS {
case "netbsd", "windows":
	continue
case "freebsd":
	if runtime.GOOS != "linux" {
		continue
	}
}

 I get the error 

fatal error: sys/endian.h: No such file or directory

when I do make test.